src/main/java/com/devonfw/ide/sonarqube/common/impl/check/naming/DevonNamingConventionClassExtendsClassCtoCheck.java

```diff
@@ -19,4 +21,10 @@ public DevonNamingConventionClassExtendsClassCtoCheck() {
     super(".*Cto");
   }
 
+  @Override
+  protected String checkDependency(JavaType source, JavaType target) {
+
+    return null;
+  }
```

Comment:
Why did we have to introduce many of these "empty" `checkDependency` methods? The method was already abstract before. Seems I am missing something. However, if we have to do this, we could instead return `null` instead of declaring it `abstract` in the super-class then to avoid further redundancy.

Reply:
The many `checkDependency` methods all come from the naming checks. Those classes were not children of `DevonArchitectureCheck` so they didn't need to implement `checkDependency`

Reply:
I guess what we could do is differentiate between checks that scan the code and checks that scan the imports. That would result in two new abstract classes, which would both beinheriting from `DevonArchitectureCheck`. Then `checkDependency` could be declared as an abstract method inside the new superclass for import checks. 
For checks that scan both the code and the imports of a file, we can just keep `DevonArchitectureCheck` as its superclass (e.g. as is the case with `DevonArchitecture3rdPartyHibernateCheck`)